### ISSUE TITLE
Is it possible to use secret from other namespace for SecretStore.spec.provider.gitlab.auth.SecretRef

### ISSUE THREAD

**Issue by KonstantinNosov** [external-secrets/external-secrets#3791]
I'm not sure if it is bug featire request or request for clarification so not following description template.

The use case I have
1. Multiple projects
2. Each project has it's own group on gitlab and own team
3. Each project has namespace on k8s
4. Gitlab variables of the project group to be consumed with external secrets gitlab provider for consumption in project namespace
5. No project may use secrets of another project

Ideally I'd like to have SecretStore pointed to project gitlab group vars in each project namespace
But it looks  like I need to create gitlab access token for each project group and secret with it in each project namespace to be referenced by SecretStore. 

It looks redundant, having a single gitlab token with access to all projects and secret for it stored out of project namespaces looks quite straighforward solution but unlike ClusterSecretStore , namespace scoped SecretStore ignores namespace field in SecretRef

Am I missing something and there is a reasoning behind such restriction or it is just smth that is not impelmented yet?

**Comment by Skarlso** [comment]
Yes.

```go
// A reference to a specific 'key' within a Secret resource,
// In some instances, `key` is a required field.
type SecretKeySelector struct {
	// The name of the Secret resource being referred to.
	Name string `json:"name,omitempty"`
	// Namespace of the resource being referred to. Ignored if referent is not cluster-scoped. cluster-scoped defaults
	// to the namespace of the referent.
	// +optional
	Namespace *string `json:"namespace,omitempty"`
	// The key of the entry in the Secret resource's `data` field to be used. Some instances of this field may be
	// defaulted, in others it may be required.
	// +optional
	Key string `json:"key,omitempty"`
}```

**Comment by KonstantinNosov** [comment]
I c.
Looks like the only option I have is using instance of ClusterSecretStore per project with   conditions[0].namespaces=[<project namespace>]

Not all project specific resources (secret store) reside in project namespace in this case, but not that critical

**Comment by Skarlso** [comment]
I'm not sure I understand. What I just showed you is the auth thing. So you just add the namespace and that's it? 😁 I'm probably misunderstanding something.

**Comment by KonstantinNosov** [comment]
Yep but  `namespace` comments has  " Ignored if referent is not cluster-scoped.". So in SecretStore only secret from same namespace may be referenced

**Comment by Skarlso** [comment]
Ah okay, sorry. I didn't notice that.
